addons/website_sale/views/templates.xml

```diff
@@ -1745,4 +1725,36 @@
         </div>
     </template>
 
+    <!-- Products Search Bar input-group template -->
+    <template id="website_sale_search_input_group" name="Products Search Bar Input">
+        <div class="input-group o_products_search_bar_input">
+            <input type="text" class="form-control o_search_products" placeholder="Search for products" data-display-description="true" data-display-price="true" data-user-filter="true" data-display-image="true" data-order-by="order_by_name_AZ"/>
+            <div class="input-group-append">
+                <button class="btn btn-outline-secondary dropdown-toggle" type="button" id="o_button_search_filter" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Search filter</button>
+                <div class="dropdown-menu" aria-labelledby="o_button_search_filter">
+                    <a class="dropdown-item js_order_by" href="#" data-value="order_by_name_AZ">Order by name (A-Z)</a>
+                    <a class="dropdown-item js_order_by" href="#" data-value="order_by_name_ZA">Order by name (Z-A)</a>
+                    <a class="dropdown-item js_order_by" href="#" data-value="order_by_price_low">Order by price: low to high</a>
+                    <a class="dropdown-item js_order_by" href="#" data-value="order_by_price_high">Order by price: high to low</a>
+                </div>
+            </div>
+        </div>
+    </template>
+
+    <!-- Products Search Bar snippet template -->
+    <template id="s_products_searchbar" name="Search Bar">
```

Comment:
You still have only one snippet, in the feature section which is only droppable as a whole big snippet. What I had in mind is a small snippet in the content section (like card, etc) which is droppable everywhere in other snippets. After discussion with JKE, why not doing both: a small and a big one (and the big one can contain the small one). Also do not use anything else than `col-lg-*` in snippets (not md, etc) because the other ones are not editable for now.